File: geocoding.php
<?php
class Geocoding {

	public function request($data) {
		$base_url = "https://geocoding.geo.census.gov/geocoder/geographies/address?benchmark=Public_AR_Census2010&format=json&vintage=Census2010_Census2010&layers=14&";
		$extension = http_build_query($data);
		$url = $base_url . $extension;
		$ch = curl_init($url);
		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
		$resp = curl_exec($ch);
		curl_close($ch);
		return json_decode($resp, true);
	}
}

$x =  new Geocoding();
$resp = $x->request(['street' => '5619 Belarbor Street', 'city' => 'Houston', 'state' => 'TX', 'zip' => '77033']);
?>

File: geocoding.R
library(httr)
#This is a function that geocodes addresses using the GeoCoding Census API. 
geocoding_api <- function(street, city, state, zip) {
  base_url = "https://geocoding.geo.census.gov/geocoder/geographies/address?benchmark=Public_AR_Census2010&format=json&vintage=Census2010_Census2010&layers=14&"
  street = paste("&street=", street, sep = "")
  city = paste("&city=", city, sep = "")
  state = paste("&state=", state, sep = "")
  zip = paste("&zip=", zip, sep = "")
  #Concatenates each command to the 
  req_url = paste(base_url, street, city, state, zip, sep = "")
  req <- GET(req_url)
  resp <- content(req)
  location_data = data.frame(
    x = resp$result$addressMatches[[1]]$coordinates$x,
    y = resp$result$addressMatches[[1]]$coordinates$y,
    geo.id = resp$result$addressMatches[[1]]$geographies$`Census Blocks`[[1]]$GEOID
  )
  location_data
}
x <- geocoding_api("5619+Belarbor+Street","Houston","TX","77033")

